lib/gs-app-list-private.h

```diff
@@ -45,16 +45,13 @@ typedef enum {
  * Flags to use when filtering. The priority of eash #GsApp is used to choose
  * which application object to keep.
  **/
-typedef enum {
-	GS_APP_LIST_FILTER_FLAG_NONE		= 0,
-	GS_APP_LIST_FILTER_FLAG_KEY_ID		= 1 << 0,
-	GS_APP_LIST_FILTER_FLAG_KEY_SOURCE	= 1 << 1,
-	GS_APP_LIST_FILTER_FLAG_KEY_VERSION	= 1 << 2,
-	GS_APP_LIST_FILTER_FLAG_PREFER_INSTALLED= 1 << 3,
-	GS_APP_LIST_FILTER_FLAG_KEY_ID_PROVIDES	= 1 << 4,
-	/*< private >*/
-	GS_APP_LIST_FILTER_FLAG_LAST
-} GsAppListFilterFlags;
+#define GS_APP_LIST_FILTER_FLAG_NONE             ((guint64) 0)
+#define GS_APP_LIST_FILTER_FLAG_KEY_ID           ((guint64) 1 << 0)
+#define GS_APP_LIST_FILTER_FLAG_KEY_SOURCE       ((guint64) 1 << 1)
+#define GS_APP_LIST_FILTER_FLAG_KEY_VERSION      ((guint64) 1 << 2)
+#define GS_APP_LIST_FILTER_FLAG_PREFER_INSTALLED ((guint64) 1 << 3)
+#define GS_APP_LIST_FILTER_FLAG_KEY_ID_PROVIDES  ((guint64) 1 << 4)
+typedef guint64 GsAppListFilterFlags;
```

Comment:
Another fix might be to add another private enum member, `GS_APP_LIST_FILTER_FLAG_FORCE_UINT64 = G_MAXUINT64` at the end of the enum, to force it to be 64-bits wide? Personally I like enums more than I like preprocessor macros.

Reply:
Really the people to ask would be kalev lember or richard hughes, upstream – it's their taste that matters :-)

Reply:
Given that we dont have much time to integrate this to 3.7.0 I would go with the macros and then push the changes upstream and we can update again in a future version depending on upstream feedback. Note that other flags are also using macros, see https://github.com/endlessm/gnome-software/commit/31c9dd212156afbb2422188cb59ee8acb5a0719f.

I will update the typo though.

Reply:
Oh, I see that patch made an analogous change from enum to `#define`. I guess that answers the What Would Upstream Do question!

Reply:
Upstreamed using the original `enum` as https://gitlab.gnome.org/GNOME/gnome-software/merge_requests/343. I think that https://github.com/endlessm/gnome-software/commit/31c9dd212156afbb2422188cb59ee8acb5a0719f is a red herring because it was written by Robert Ancell, so it shows us What Upstream Will Accept, not What Upstream Would Do (or What Upstream Should Do, if I’m on my high horse).